### ISSUE TITLE
Multiple instances of the same Reusable Q&A on the same page results in a duplicate ID (508-4)

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#11448]
## Description
There is a use case for allowing editors to use the same Q&A in multiple places on the same page (either an FAQ page or Resources & Support page), however when this happens, it results in a duplicate ID on the front end of the page.

### Use case
[Example page](https://www.va.gov/resources/the-pact-act-and-your-va-benefits/)
- There are 2 section on this page, one for Gulf War Veterans and one for Vietnam Veterans.  Each section has the same Q&A ("Are there more exposure-related presumptive conditions?") because each of these section of Veterans needs this info.  This is where the duplicate ID is coming into play.
![image](https://user-images.githubusercontent.com/106678594/199812357-5fbe4bef-cedc-434b-8afe-29c5ea079c9f.png)

### Additional Context
Templates for this work will be in the `content-build` repo.

I spoke to Swirt about this and he mentioned that this should be able to be handled programmatically and be he believes there is already a pattern somewhere on the front end, though he wasn't sure where, that could handle this.

He said that there should never be the same Q&A within the same group, which is correct.  So if we could combine the ID for the group with the ID for the Q&A element, that would result in a unique ID for each and resolve the issue.

### Where reusable Q&As appear

* FAQ pages
* Resources & Support pages

Multiple Q&As can also appear on other content types, but **not** as Reusable Q&A nodes. (As paragraphs, etc. instead)

## Acceptance Criteria
- [x] Update generated ID to combine group ID with element ID
- [x] Fix here doesn't worse the problem in #11354,   (because reusable Q&As can be created as a group of Accordions on FAQ content, e.g.)
    * https://prod.cms.va.gov/node/8233/edit
    * https://www.va.gov/resources/verifying-your-identity-on-vagov/
- [ ] Accessibility Review

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by laflannery** [comment]
@wesrowe I believe this falls under Public Websites - this is not something that is a very high priority at all.  It is however being reported in our daily accessibility scan and I am just ignoring it for now but if this is something that could fit into a relatively soon-ish sprint (next 1 or 2 maybe?) that would be helpful.

Let me know if you want to chat about this or need any additional information from me.

**Comment by wesrowe** [comment]
@ryguyk, I assigned this to you for pre-refinement because it involves looking for patterns elsewhere in the FE. If you'd prefer to hand it off that's fine with me.

**Comment by laflannery** [comment]
@wesrowe I'm doing some clean up/prep of my 508 tickets.  With @ryguyk moving to Facilities will we need someone else to refine this?  Is there anything I could do to assist? I would love to get it moving so the a11y report can be cleaned up.

**Comment by wesrowe** [comment]
@laflannery, thanks for the reminder, I'll assign to someone else for refinement. It has languished because it's a Prio 4 defect and we have a pretty full backlog for FE engineers. Is the a11y report your automated report?

**Comment by laflannery** [comment]
It is the report connected to the `#-daily-accessibility-scan` Slack channel.  I'm the only one that checks this regularly at the moment and therefore I'm the only one it's annoying however @timcosgrove [found this as on his own](https://dsva.slack.com/archives/CT4GZBM8F/p1671124647267389) and I do not know for sure if this impacts his daily life, probably not or he would have thrown a bigger flag?

However, I do think that because it's actively causing an error in our scans it would be good to resolve, even though the defect level itself is low.

**Comment by wesrowe** [comment]
Interesting – I'll ask someone to estimate it, might be a small lift. 

@laflannery, do we know yet if there's a single template that needs updating? Or do we need to figure that out?

**Comment by laflannery** [comment]
@wesrowe I don't think I know enough to answer that? I want to say it would be more than one because I believe it could happen anywhere reusable content would show on the FE, we would need to redo how these IDs are generated.  Would it be helpful if I figured that out for you?  I can do some testing on stage and see if this happens in other places?

**Comment by wesrowe** [comment]
Whoever review this for pre-refinement, note that we should look to see which templates might be impacted by this same problem.

**Comment by jtmst** [comment]
This might be worth a discussion as the div IDs are derived from the entity.entityId which is passed from drupal.  Are we sure this is easier to fix on the front? `<div id="{{ fieldQA.entity.entityBundle }}-{{ fieldQA.entity.entityId }}">`

**Comment by laflannery** [comment]
@wesrowe I just wanted to check back in on this - we are approaching the Platform SLA for a defect 4 ( Trivial. Consider fixing or exploring in 2-4 sprints post-launch.) so I wanted to see if we could get this into the next sprint to start looking at. Thanks!

**Comment by wesrowe** [comment]
@jtmst, re your comment above, I think the details in the ticket (Swirt's suggestion I think) of combining group and q&q IDs might address the concern. Do they not? And whom would you pull in to clear up the FE/BE question?

**Comment by jilladams** [comment]
Front end refinement: 
* Combining group ID + entity ID = should be possible in front-end. 
* We want to confirm with Drupal folks if this is for sure not possible to address in Drupal > GraphQL

Re: which templates: 
* FAQ multiple 
* R&S

**Comment by randimays** [comment]
The changes have been merged into `content-build` and have been validated in staging. They will be deployed to production tomorrow afternoon and I will validate there as well. This is ready for an accessibility review from @laflannery when we she returns from PTO next week.

**Comment by laflannery** [comment]
@randimays I confirmed this is resolved on Prod and no longer being flagged in my a11y scan, thank you!

**Comment by randimays** [comment]
Thank you @laflannery! @wesrowe Moved into PO review.

**Comment by wesrowe** [comment]
If Laura approves then I do too. Closing.
